src/bin/ion/commands/beta/generate/templates/rust/struct.templ

```diff
@@ -1,128 +1,139 @@
+                writer.step_in(IonType::Struct)?;
+                {% for field in fields %}
+                writer.set_field_name("{{ field.name }}");
+                    {% if field.value_type | is_built_in_type == false %}
+                        self.{{ field.name | snake }}.write_to(writer)?;
+                    {% else %}
+                        {# TODO: Change the following `to_owned` to only be used when writing i64,f32,f64,bool which require owned value as input #}
+                        writer.write_{% if field.isl_type_name == "symbol" %}symbol{% else %}{{ field.value_type | lower }}{% endif %}(self.{{ field.name | snake }}.to_owned())?;
+                    {% endif %}
+                {% endfor %}
+                writer.step_out()?;
+            {% elif abstract_data_type is object and abstract_data_type is containing("Sequence")  %}
+                {% if abstract_data_type["Sequence"].sequence_type == "List" %}
+                    writer.step_in(IonType::List)?;
                 {% else %}
-                    {# TODO: Change the following `to_owned` to only be used when writing i64,f32,f64,bool which require owned value as input #}
-                    writer.write_{% if field.isl_type_name == "symbol" %}symbol{% else %}{{ field.value_type | lower }}{% endif %}(self.{{ field.name | snake }}.to_owned())?;
+                    writer.step_in(IonType::SExp)?;
                 {% endif %}
-            {% endfor %}
-            writer.step_out()?;
-        {% elif abstract_data_type is object and abstract_data_type is containing("Sequence")  %}
-            {% if abstract_data_type["Sequence"].sequence_type == "List" %}
-                writer.step_in(IonType::List)?;
-            {% else %}
-                writer.step_in(IonType::SExp)?;
+                for value in &self.value {
+                    {% if abstract_data_type["Sequence"].element_type | is_built_in_type  == false %}
+                        value.write_to(writer)?;
+                    {% else %}
+                       writer.write_{% if fields[0].isl_type_name == "symbol" %}symbol{% else %}{{ abstract_data_type["Sequence"].element_type | lower }}{% endif %}(value.to_owned())?;
+                    {% endif %}
+                }
+                writer.step_out()?;
             {% endif %}
-            for value in &self.value {
-                {% if abstract_data_type["Sequence"].element_type | is_built_in_type  == false %}
-                    value.write_to(writer)?;
-                {% else %}
-                   writer.write_{% if fields[0].isl_type_name == "symbol" %}symbol{% else %}{{ abstract_data_type["Sequence"].element_type | lower }}{% endif %}(value.to_owned())?;
-                {% endif %}
-            }
-            writer.step_out()?;
-        {% endif %}
-        Ok(())
+            Ok(())
+        }
     }
+
+    {% for inline_type in anonymous_types -%}
```

Comment:
(🗺️ PR tour) This is where all the anonymous types will be generated for given top level type definition.